coderd/httpmw/prometheus.go

```diff
@@ -98,3 +99,31 @@ func Prometheus(register prometheus.Registerer) func(http.Handler) http.Handler
 		})
 	}
 }
+
+func getRoutePattern(r *http.Request) string {
+	rctx := chi.RouteContext(r.Context())
+	if rctx == nil {
+		return ""
+	}
+
+	if pattern := rctx.RoutePattern(); pattern != "" {
+		// Pattern is already available
+		return pattern
+	}
+
+	routePath := r.URL.Path
+	if r.URL.RawPath != "" {
+		routePath = r.URL.RawPath
+	}
+
+	tctx := chi.NewRouteContext()
+	routes := rctx.Routes
+	if routes != nil && routes.Match(tctx, r.Method, routePath) {
+		// No matching pattern, so just return an empty string.
+		// It is done to avoid returning a static path for frontend requests.
+		return ""
```

Comment:
Could this differentiate between "static" and "unknown API request"? If the path begins with `/api/` then it's an unknown API request, otherwise it's a static request (all non-API 404s serve HTML)